### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by brainskev**
In this branch I achieved the following;

- Created model classes for all entities
- Set up associations between models.
- Used models to insert data

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_posts_counter
+    author.increment!(:post_counter)
+  end
```
Comment (Abenezer-Tilahun) [inline app/models/post.rb:8]: - [ Optional ] Kindly I recommend you  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private Just like this :- 

  # app/models/post.rb
```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (Abenezer-Tilahun) [inline app/models/comment.rb:7]: - [ Optional ] Kindly I recommend you  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counter method every time a new comment has been saved to database, then make the update_comments_counter method private Just like this :- 

  # app/models/comment.rb
```
  after_save :update_comments_counter

  private

  def update_comments_counter
    post.increment!(:comments_counter)
  end
```

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @brainskev 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Blog App - processing data in models.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

File: app/models/comment.rb (hunk 2 above)
Comment (brainskev) [reply app/models/comment.rb:7]: Hi @Abenezer-Tilahun 

Just wanted to drop a quick note to say thank you for reviewing my code. Your feedback and suggestions have been incredibly valuable, and I truly appreciate your time and expertise. I would like to connect with you on [GitHub ](https://github.com/brainskev) & [LinkedIn](https://www.linkedin.com/in/kelvinkaviku/) to stay in touch and potentially collaborate on future projects. Your contributions have been invaluable, and I believe we can learn a lot from each other.

Looking forward to connecting!

Best regards,
Kelvin Kaviku 
@brainskev

**Pull request merged**